pkg/controller/clusterautoscaler/clusterautoscaler.go

```diff
@@ -109,6 +110,17 @@ func AutoscalerArgs(ca *v1.ClusterAutoscaler, cfg *Config) []string {
 
 	if ca.Spec.BalanceSimilarNodeGroups != nil {
 		args = append(args, BalanceSimilarNodeGroupsArg.Value(*ca.Spec.BalanceSimilarNodeGroups))
+
+		switch strings.ToLower(cfg.CloudProvider) {
+		case "alibabacloud":
+			args = append(args, BalancingIgnoreLabelArg.Value("topology.diskplugin.csi.alibabacloud.com/zone"))
+		case "aws":
+			args = append(args, BalancingIgnoreLabelArg.Value("topology.ebs.csi.aws.com/zone"))
+		case "azure":
+			args = append(args, BalancingIgnoreLabelArg.Value("topology.disk.csi.azure.com/zone"))
+		case "gcp":
+			args = append(args, BalancingIgnoreLabelArg.Value("topology.gke.io/zone"))
```

Comment:
Wonder if we should make the various cloud provider label names constants that we can group per platform? I feel like if we need to add more this could become confusing if they are embedded on individual lines

Reply:
+1, i think that's a nice upgrade